src/vm/methodtablebuilder.cpp

```diff
@@ -4085,8 +4089,10 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
                         // Non-value-classes cannot contain by-ref-like instance fields
                         BuildMethodTableThrowException(IDS_CLASSLOAD_BYREFLIKE_NOTVALUECLASSFIELD);
                     }
-
-                    bmtFP->fIsByRefLikeType = true;
+                    if (!bmtFP->fIsByRefLikeType)
```

Comment:
Maybe update the comment above (`// Inherit IsByRefLike characteristic from fields`) since we're not inheriting it anymore, but merely checking?